Statistics/Histograms/ThreadsafeInlineHistogram.cs

```diff
@@ -461,6 +461,10 @@ public void AddObservationsToHistogram(double[] data)
         public int FindBinCount(double x, bool cumulative = true)
         {
             int obsIndex = Convert.ToInt32(Math.Floor((x - _Min) / _BinWidth));
+            if (obsIndex == _BinCounts.Length)
```

Comment:
im a bit surprised here, floor should do that operation, but i assume you hit a bug, so this makes sense.